src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs

```diff
@@ -379,6 +356,62 @@ public bool ExecuteReplicationOnce(TcpConnectionOptions tcpConnectionOptions, Ou
             }
         }
 
+        private bool CanContinueBatch(ReplicationState state, ref long next)
+        {
+            if (MissingAttachmentsInLastBatch)
```

Comment:
Isn't this called for any item? In that case, what is the meaning of removing it one at a time?

Reply:
this code called only when `item.TransactionMarker != replicationState.LastTransactionMarker`
then we try to remove `replicationState.LastTransactionMarker`, if there are `state.MissingTxMarkers` we have to continue the replication batch, else we allow to interrupt it if needed